src/main/java/mate/academy/security/jwt/JwtTokenProvider.java

```diff
@@ -0,0 +1,98 @@
+package mate.academy.security.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.Jws;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.SignatureAlgorithm;
+import io.jsonwebtoken.security.Keys;
+import java.security.Key;
+import java.util.Date;
+import java.util.List;
+import java.util.Optional;
+import javax.annotation.PostConstruct;
+import javax.servlet.http.HttpServletRequest;
+import mate.academy.exception.InvalidJwtAuthenticationException;
+import org.springframework.core.env.Environment;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.stereotype.Component;
+
+@Component
+public class JwtTokenProvider {
+    private static final Long DEFAULT_VALIDITY_IN_MILLISECONDS = 360000000000000000L;
+    private static final String BEARER_PREFIX = "Bearer";
+    private static final String AUTHORIZATION_HEADER = "Authorization";
+    private final UserDetailsService userDetailsService;
+    private final Environment environment;
+    private Key secretKey;
+    private Long validityInMilliseconds;
```

Comment:
we can use `@Value("${jwt-validity-time:360000000000000000}")`
and `@Value("${jwt-secret-secretKey:default_secret}")`